src/core/lib/channel/handshaker.cc

```diff
@@ -22,11 +22,15 @@
 
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>
+#include <grpc/support/string_util.h>
 
 #include "src/core/lib/channel/channel_args.h"
 #include "src/core/lib/channel/handshaker.h"
+#include "src/core/lib/debug/trace.h"
 #include "src/core/lib/iomgr/timer.h"
 
+grpc_core::TraceFlag grpc_handshaker_trace(false, "handshaker");
```

Comment:
I'm guessing the DebugOnlyTraceFlag will not work?

Reply:
I'm sure it would work, but I don't know why it would be necessary.  In general, DebugOnlyTraceFlag is for cases where enabling the tracer imposes a large amount of overhead that we don't want in opt builds (e.g., things like ref-count tracing, which occurs a *lot*).  That's not the case here, so there's no reason not to allow this tracer even in opt builds.

Reply:
I agree that the overhead would not be much but I would prefer to not add unnecessary overhead.

Reply:
By that logic, why would we support *any* tracer in opt mode?  I don't understand why this particular tracer would be different than any other.

Note that there is no additional overhead unless the tracer is actually enabled.

Reply:
Mark, is there a specific reason to want this tracer in opt builds ? We have seen a general downward drift in performance benchmarks, so I would lean towards avoiding  this overhead unless there is a specific need.

Reply:
As per our discussion, it is very useful to support tracers in opt mode, because this allows customers to enable them without pushing a debug binary (which they often can't for performance reasons).  This makes it easier to gather useful debugging information when a customer runs into a problem.

If we're really worried about the branch-prediction performance impact of a few `if` statements checking whether the tracer is enabled, I would suggest that we find a structural way to make use of the `GPR_UNLIKELY` macro that @yashykt recently added in #15009.  For example, perhaps we could use that in the `TraceFlag::enabled()` method.

Reply:
Based on offline discussion, we will explore adding compiler hints to the tracers to avoid perf impact.